#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_makify.dpatch by  <seneca@localhost.localdomain>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adds Makefiles

@DPATCH@

diff -Naur ledgersmb/bin/Makefile ledgersmb-1.1.99beta1-new/bin/Makefile
--- ledgersmb/bin/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/bin/Makefile	2006-11-13 17:11:04.000000000 -0500
@@ -0,0 +1,13 @@
+PACKAGE = ledgersmb
+
+FILES = $(shell ls *.pl)
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/ledgersmb/bin/custom
+	cp $(FILES) $(DESTDIR)/usr/share/ledgersmb/bin
+	find $(DESTDIR)/usr/share/ledgersmb/bin -type d -exec chmod 755 '{}' \;
+	find $(DESTDIR)/usr/share/ledgersmb/bin -type f -exec chmod 644 '{}' \;
diff -Naur ledgersmb/contrib/Makefile ledgersmb-1.1.99beta1-new/contrib/Makefile
--- ledgersmb/contrib/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/contrib/Makefile	2006-11-13 17:14:01.000000000 -0500
@@ -0,0 +1,9 @@
+PACKAGE = ledgersmb
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/doc/ledgersmb/contrib
+	cp -r * $(DESTDIR)/usr/share/doc/ledgersmb/contrib
diff -Naur ledgersmb/css/Makefile ledgersmb-1.1.99beta1-new/css/Makefile
--- ledgersmb/css/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/css/Makefile	2006-11-13 17:02:56.000000000 -0500
@@ -0,0 +1,13 @@
+PACKAGE = ledgersmb
+
+FILES = $(shell ls ledger-smb*.css)
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/var/lib/ledgersmb/css
+	cp $(FILES) $(DESTDIR)/var/lib/ledgersmb/css
+	chmod 644 $(DESTDIR)/var/lib/ledgersmb/css/*
+	ln -sf /var/lib/ledgersmb/css $(DESTDIR)/usr/share/ledgersmb
diff -Naur ledgersmb/doc/Makefile ledgersmb-1.1.99beta1-new/doc/Makefile
--- ledgersmb/doc/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/doc/Makefile	2006-11-13 17:58:52.000000000 -0500
@@ -0,0 +1,18 @@
+PACKAGE = ledgersmb
+
+FILES = $(shell ls | grep -v Makefile | grep -v COPYING | grep -v COPYRIGHT)
+SUBDIRS = API manual database samples
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/ledgersmb/doc
+	cp LedgerSMB-manual.pdf $(DESTDIR)/usr/share/ledgersmb/doc
+	mkdir -p $(DESTDIR)/usr/share/doc/ledgersmb
+	cp -a $(FILES) $(DESTDIR)/usr/share/doc/ledgersmb
+	cp -a API $(DESTDIR)/usr/share/doc/ledgersmb
+	cp -a manual $(DESTDIR)/usr/share/doc/ledgersmb
+	cp -a database $(DESTDIR)/usr/share/doc/ledgersmb
+	cp -a samples $(DESTDIR)/usr/share/doc/ledgersmb
diff -Naur ledgersmb/drivers/Makefile ledgersmb-1.1.99beta1-new/drivers/Makefile
--- ledgersmb/drivers/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/drivers/Makefile	2006-11-13 17:12:33.000000000 -0500
@@ -0,0 +1,10 @@
+PACKAGE = ledgersmb
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/ledgersmb/drivers
+	cp *.pl $(DESTDIR)/usr/share/ledgersmb/drivers
+	chmod 644 $(DESTDIR)/usr/share/ledgersmb/drivers/*
diff -Naur ledgersmb/LedgerSMB/Makefile ledgersmb-1.1.99beta1-new/LedgerSMB/Makefile
--- ledgersmb/LedgerSMB/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/LedgerSMB/Makefile	2006-11-13 18:03:13.000000000 -0500
@@ -0,0 +1,16 @@
+PACKAGE = ledgersmb
+
+FILES := $(shell ls *.pm)
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/ledgersmb/LedgerSMB
+	cp -r $(FILES) $(DESTDIR)/usr/share/ledgersmb/LedgerSMB
+	cp -r Taxes $(DESTDIR)/usr/share/ledgersmb/LedgerSMB
+	cp -r Session $(DESTDIR)/usr/share/ledgersmb/LedgerSMB
+	cp -r CreditCard $(DESTDIR)/usr/share/ledgersmb/LedgerSMB
+	find $(DESTDIR)/usr/share/ledgersmb/LedgerSMB -type f -exec chmod 644 '{}' \;
+	find $(DESTDIR)/usr/share/ledgersmb/LedgerSMB -type d -exec chmod 755 '{}' \;
diff -Naur ledgersmb/locale/Makefile ledgersmb-1.1.99beta1-new/locale/Makefile
--- ledgersmb/locale/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/locale/Makefile	2006-11-13 16:56:15.000000000 -0500
@@ -0,0 +1,21 @@
+PACKAGE = ledgersmb
+
+LOCALES = $(shell ls po/*)
+
+localedir = /usr/share/ledgersmb/locale
+dlocaledir = $(DESTDIR)$(localedir)
+srctopdir := $(shell pwd)/..
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(dlocaledir)/po
+	mkdir -p $(dlocaledir)/html
+	cp po/* $(dlocaledir)/po
+	cp html/*.html $(dlocaledir)/html
+	cp LedgerSMB.pot $(dlocaledir)
+	cp *.html $(dlocaledir)
+	find $(dlocaledir) -type d -exec chmod 755 '{}' \;
+	find $(dlocaledir) -type f -exec chmod 644 '{}' \;
diff -Naur ledgersmb/Makefile ledgersmb-1.1.99beta1-new/Makefile
--- ledgersmb/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/Makefile	2006-11-13 18:19:16.000000000 -0500
@@ -0,0 +1,36 @@
+PACKAGE = ledgersmb
+
+SUBDIRS = bin doc drivers LedgerSMB locale sql templates css contrib utils
+
+LIB_FILES = favicon.ico menu.ini \
+	    ledgersmb.conf.default VERSION index.html \
+	    ledger-smb.png ledger-smb.gif ledger-smb_small.png \
+	    $(shell ls *.pl | grep -v -E '(pos\.conf)\.pl')
+
+CFG_FILES = ledgersmb.conf ledgersmb-httpd.conf pos.conf.pl
+
+all:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/ledgersmb
+	cp $(LIB_FILES) $(DESTDIR)/usr/share/ledgersmb
+
+	set -e ; for subdir in $(SUBDIRS) ; do \
+	  $(MAKE) -C $$subdir $@ ; \
+	done
+
+	mkdir -p $(DESTDIR)/etc/ledgersmb
+	cp $(CFG_FILES) $(DESTDIR)/etc/ledgersmb
+	cd $(DESTDIR)/usr/share/ledgersmb ; \
+	ln -sf /etc/ledgersmb/pos.conf.pl ; \
+	ln -sf /etc/ledgersmb/images
+	mkdir -p $(DESTDIR)/var/lib/ledgersmb/spool
+	chown www-data:www-data $(DESTDIR)/var/lib/ledgersmb/spool
+
+	mkdir -p $(DESTDIR)/etc/ledgersmb/images
+	mkdir -p $(DESTDIR)/etc/ledgersmb/images/demo
+	cp $(DESTDIR)/etc/ledgersmb/templates/demo/*.[^ht]* $(DESTDIR)/etc/ledgersmb/images
+	cp $(DESTDIR)/etc/ledgersmb/templates/demo/*.[^ht]* $(DESTDIR)/etc/ledgersmb/images/demo
+	chown -R www-data:www-data $(DESTDIR)/etc/ledgersmb/templates
+	chgrp www-data $(DESTDIR)/etc/ledgersmb/ledgersmb.conf
+	chmod 640 $(DESTDIR)/etc/ledgersmb/ledgersmb.conf
+
+clean:
diff -Naur ledgersmb/sql/Makefile ledgersmb-1.1.99beta1-new/sql/Makefile
--- ledgersmb/sql/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/sql/Makefile	2006-11-13 17:06:48.000000000 -0500
@@ -0,0 +1,14 @@
+PACKAGE = ledgersmb
+
+FILES = *.sql
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/ledgersmb/sql/legacy
+	cp $(FILES) $(DESTDIR)/usr/share/ledgersmb/sql
+	cp -r legacy $(DESTDIR)/usr/share/ledgersmb/sql
+	find $(DESTDIR)/usr/share/ledgersmb/sql -type f -exec chmod 644 '{}' \;
+	find $(DESTDIR)/usr/share/ledgersmb/sql -type d -exec chmod 755 '{}' \;
diff -Naur ledgersmb/templates/Makefile ledgersmb-1.1.99beta1-new/templates/Makefile
--- ledgersmb/templates/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/templates/Makefile	2006-11-13 17:11:20.000000000 -0500
@@ -0,0 +1,16 @@
+PACKAGE = ledgersmb
+
+FILES = *.html *.tex *.txt
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/etc/ledgersmb/templates
+	cp -a $(FILES) $(DESTDIR)/etc/ledgersmb/templates
+	cp -a demo $(DESTDIR)/etc/ledgersmb/templates
+	chown -R www-data:www-data $(DESTDIR)/etc/ledgersmb/templates
+	ln -sf /etc/ledgersmb/templates $(DESTDIR)/usr/share/ledgersmb/
+	find $(DESTDIR)/etc/ledgersmb/templates -type d -exec chmod 755 '{}' \;
+	find $(DESTDIR)/etc/ledgersmb/templates -type f -exec chmod 644 '{}' \;
diff -Naur ledgersmb/utils/Makefile ledgersmb-1.1.99beta1-new/utils/Makefile
--- ledgersmb/utils/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ ledgersmb-1.1.99beta1-new/utils/Makefile	2006-11-13 17:14:17.000000000 -0500
@@ -0,0 +1,9 @@
+PACKAGE = ledgersmb
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/doc/ledgersmb/utils
+	cp -r * $(DESTDIR)/usr/share/doc/ledgersmb/utils
